src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1952,6 +1953,12 @@ namespace winrt::TerminalApp::implementation
 
         layout.InitialSize(windowSize);
 
+        // We don't actually know our own position. So we have to ask the window
+        // layer for that.
+        const auto launchPosRequest{ winrt::make<LaunchPositionRequest>() };
+        RequestLaunchPosition.raise(*this, launchPosRequest);
+        layout.InitialPosition(launchPosRequest.Position());
```

Comment:
BTW as I mentioned briefly in #12633 we should probably use `GetWindowPlacement` here. It only needs the `HWND` which the `TerminalPage` already knows about. The struct could then be serialized to a JSON object. It even got `WPF_ASYNCWINDOWPLACEMENT` which may be useful for us: https://learn.microsoft.com/en-us/windows/win32/api/winuser/ns-winuser-windowplacement

Reply:
Yes, though, we may also want to be careful about how that plays with focus mode too. Since forever, session restore + fomo has restored at the wrong size, I think wrongly including the size of the tab row. (I cannot for the life of me find that issue though)